pkg/secretless/entrypoint/entrypoint.go

```diff
@@ -48,13 +48,19 @@ func StartSecretless(params *SecretlessOptions) {
 		log.Fatalln(err)
 	}
 
-	// Prepare Service
-	secretless := proxyservice.NewProxyServices(cfg, availPlugins, logger, evtNotifier)
-	signal.StopOnExitSignal(secretless)
-
 	handlePerformanceProfiling(params.ProfilingMode)
 
-	secretless.Start()
+	// Start Services
+	allServices := proxyservice.NewProxyServices(cfg, availPlugins, logger, evtNotifier)
+	signal.StopOnExitSignal(allServices)
+
+	err = allServices.Start()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// Block until we receive an exit signal
+	<- signal.NewExitSignalChan()
```

Comment:
There's a race condition here. The service cleanup will never happen because once line 63 is hit the program will terminate. We'll need some way to coordinate multiple consumers of these exit signals.

Reply:
Rewrote the signal package entirely, should be fixed now.